lib/mox.ex

```diff
@@ -891,4 +910,83 @@ defmodule Mox do
       pids when is_list(pids) -> pids
     end
   end
+
+  ## Ownership
+
+  @doc false
+  def start_link_ownership do
+    case N.start_link(name: @ownership_server) do
+      {:error, {:already_started, _}} -> :ignore
+      other -> other
+    end
+  end
+
+  defp add_expectation(owner_pid, {mock, _, _} = key, expectation) do
+    # First, make sure that the owner_pid is either the owner or that the mock
+    # isn't owned yet. Otherwise, return an error.
+    case N.fetch_owner(@ownership_server, [owner_pid], mock, @timeout) do
+      {tag, ^owner_pid} when tag in [:ok, :shared_owner] -> :ok
+      {:shared_owner, other_owner} -> throw({:error, {:not_shared_owner, other_owner}})
+      {:ok, other_owner} -> throw({:error, {:currently_allowed, other_owner}})
```

Comment:
Let's not use throw? If we encapsulate this into a private function, using `with` should be straight-forward.